tvb/analyzers/fragility.py

```diff
@@ -0,0 +1,172 @@
+# -*- coding: utf-8 -*-
+#
+#
+#  Fragility-Algorithm Package. 
+#
+#   CITATION:
+# When using for scientific publications, please cite it as follows:
+#
+#   <INCLUDE CITATION>
+#
+
+"""
+Perform Fragility Analysis on a TimeSeries Object and returns an
+Fragility datatype.
+
+.. moduleauthor:: Adam Li <adam2392@gmail.com>
+
+"""
+
+import numpy
+import tvb.datatypes.time_series as time_series
+import tvb.datatypes.arrays as arrays
+import tvb.basic.traits.core as core
+import tvb.basic.traits.types_basic as basic
+from tvb.basic.logger.builder import get_logger
+
+from .fragility_lib.preprocess.filters import FilterLinearNoise
+from .fragility_lib.execute.singlecore.singlemvar import SingleMvar
+from .fragility_lib.execute.singlecore.singlepert import SinglePert
+from .fragility_lib.execute.singlenode.parallelmvar import ParallelMvar
+from .fragility_lib.execute.singlenode.parallelpert import ParallelPert
+
+import psutil
+
+LOG = get_logger(__name__)
+
+class mvarwindowed(core.Type):
+    """
+    Takes a TimeSeries datatype(x) and returns a sequence of mvar models (A)
+
+    :math: x(t+1) = A*x(t)
+
+    MVAR takes time-points as observations and nodes as variables
+
+    It uses a sliding window algorithm to estimate the mvar models
+    """
+
+    timeseries = time_series.TimeSeries(
+        label="Time Series",
+        required=True,
+        doc="The timeseries to which the fragility analysis is to be applied.")
+
+    winsize = basic.Integer(
+        label="Size of the windows to use in milliseconds.",
+        required=True,
+        default=250,
+        doc="Size of the windows to use in milliseconds.")
+
+    stepsize = basic.Integer(
+        label="Size of the windows to use in milliseconds.",
+        required=True,
+        default=125,
+        doc="Size of the windows to use in milliseconds.")
+
+    samplerate = basic.Float(
+        label="Sample rate of signal in Hz.",
+        required=True,
+        default=1000,
+        doc="Sample rate of the signal in Hz")
+
+    # whether or not to use multiprocessing
+    parallelize = False
+    numcores = psutil.cpu_count() / 2.
```

Comment:
`psutil` is a soft dependency, please use `multiprocessing.cpu_count()`, which ships with Python

Reply:
Fixed.